Automattic/jetpack
# Record Meter: don't use last_indexed_date to decide if a site is indexed or not

We currently rely on the `last_indexed_date` from the API to decide if a site is indexed or not. During the recent call-for-testing, @jeherve showed us an API response for a site that clearly had records, but `last_indexed_date` was null.

![image](https://user-images.githubusercontent.com/17325/177472976-897f62bd-5832-49b4-bf9e-e7a17a4d62f6.png)

#### Changes proposed in this Pull Request:
This PR removes `last_indexed_date` from our logic when deciding if a site has been indexed.

#### Other information:


#### Jetpack product discussion
p8oabR-Tl-p2#comment-6400

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
It's hard to recreate this situation without manually changing the API response, but you can check that the sign up and indexing process work as expected:

1. Create a new test site.
2. Install the Jetpack Search plugin.
3. Visit `/wp-admin/admin.php?page=jetpack-search&features=record-meter` and check that you're shown appropriate messages during the indexing process.

You can also ensure that the unit tests pass.

`cd projects/packages/search && pnpm test`
